check/src/typecheck.rs

```diff
@@ -1441,6 +1438,26 @@ fn primitive_type(op_type: &str) -> ArcType {
     }
 }
 
+/// Returns a span appropriate to mark a type error for `e`. For example the example below returns
+/// the `^` span instead of  `~`
```

Comment:
Tried to make it a bit more abstract in case this function were extended produce different spans for expressions other than `let` and `type` but I can change to to only talk about those cases.